app/src/main/java/com/google/samples/apps/sunflower/compose/PlantDetailView.kt

```diff
@@ -0,0 +1,505 @@
+import androidx.compose.Composable
+import androidx.compose.getValue
+import androidx.compose.onCommit
+import androidx.compose.remember
+import androidx.compose.setValue
+import androidx.compose.state
+import androidx.core.text.HtmlCompat
+import androidx.ui.animation.Transition
+import androidx.ui.core.Alignment
+import androidx.ui.core.ContentScale
+import androidx.ui.core.ContextAmbient
+import androidx.ui.core.DensityAmbient
+import androidx.ui.core.LayoutCoordinates
+import androidx.ui.core.Modifier
+import androidx.ui.core.ViewAmbient
+import androidx.ui.core.drawLayer
+import androidx.ui.core.drawOpacity
+import androidx.ui.core.globalPosition
+import androidx.ui.core.onPositioned
+import androidx.ui.core.semantics.semantics
+import androidx.ui.foundation.Box
+import androidx.ui.foundation.Icon
+import androidx.ui.foundation.ScrollerPosition
+import androidx.ui.foundation.Text
+import androidx.ui.foundation.VerticalScroller
+import androidx.ui.foundation.drawBackground
+import androidx.ui.foundation.shape.corner.CircleShape
+import androidx.ui.foundation.shape.corner.RoundedCornerShape
+import androidx.ui.graphics.Color
+import androidx.ui.layout.Arrangement
+import androidx.ui.layout.Column
+import androidx.ui.layout.ColumnScope.gravity
+import androidx.ui.layout.Row
+import androidx.ui.layout.Spacer
+import androidx.ui.layout.Stack
+import androidx.ui.layout.fillMaxSize
+import androidx.ui.layout.fillMaxWidth
+import androidx.ui.layout.offset
+import androidx.ui.layout.padding
+import androidx.ui.layout.preferredHeight
+import androidx.ui.layout.size
+import androidx.ui.layout.sizeIn
+import androidx.ui.layout.wrapContentSize
+import androidx.ui.livedata.observeAsState
+import androidx.ui.material.FloatingActionButton
+import androidx.ui.material.IconButton
+import androidx.ui.material.MaterialTheme
+import androidx.ui.material.Surface
+import androidx.ui.material.TopAppBar
+import androidx.ui.material.icons.Icons
+import androidx.ui.material.icons.filled.Add
+import androidx.ui.material.icons.filled.ArrowBack
+import androidx.ui.material.icons.filled.Share
+import androidx.ui.res.stringResource
+import androidx.ui.semantics.accessibilityLabel
+import androidx.ui.text.font.FontWeight
+import androidx.ui.tooling.preview.Preview
+import androidx.ui.unit.Dp
+import androidx.ui.unit.IntSize
+import androidx.ui.unit.dp
+import androidx.ui.viewinterop.AndroidView
+import androidx.ui.viewmodel.viewModel
+import com.google.android.material.snackbar.Snackbar
+import com.google.samples.apps.sunflower.R
+import com.google.samples.apps.sunflower.data.Plant
+import com.google.samples.apps.sunflower.utilities.InjectorUtils
+import com.google.samples.apps.sunflower.viewmodels.PlantDetailViewModel
+import dev.chrisbanes.accompanist.coil.CoilImageWithCrossfade
+import dev.chrisbanes.accompanist.mdctheme.MaterialThemeFromMdcTheme
+
+private const val ParallaxDelta = 2f
+private const val HeaderTransitionOffset = 150f
```

Comment:
Shouldn't this offset be in `Dp`s?

Reply:
It could, but this is used in conjunction with the scroller position which is a float

Reply:
sure, you'll need to convert it to pixels via density to use it but it feels like the threshold should be density independent just like the height.

Reply:
Yeah, my point was that this never gets converted to Dps because it compares the scrollerPosition (Float) with the original position of name on the screen (also Float)